mock-rollout/src/store/HardwareContext.tsx

```diff
@@ -53,55 +57,50 @@ const deleteHardware = (hardwares: IHardware[], id: number): IHardware[] => {
     return hardwares;
 }
 
-export const HardwareContextProvider: FC<IHardwareProps> = props => {
-    const [ getHardwareState, setHardwareState ] = useState<IState>({ hardwares: [] });
-    const [ errorMessage, setErrorMessage ] = useState<IState>({ error: '' });
-    const [ state, dispatch ] = useReducer(reducer, initialHardwareState);
-    
-    const store = { state, dispatch };
+export const HardwareContextProvider: FC<IHardwareProviderState> = (props: IHardwareProviderState) => {
+    const [ hardwareState, dispatchHardwareState ] = useReducer(reducer, initialHardwareState);
 
-    async function handleGetData() {
+    const getHardware = async () => {
         await HardwareService.getHardware()
             .then(response => {
-                setHardwareState({ hardwares: response });
-                dispatch({
+                dispatchHardwareState({
                     type: 'GET_SUCCESS', 
-                    hardwares: getHardwareState.hardwares!,
+                    hardwares: response || [],
```

Comment:
why the ! operator everywhere? there are cases when the fallback shoudl be different than undefined